sunpy/coordinates/spice.py

```diff
@@ -0,0 +1,370 @@
+        _center_registry = {'SOLAR SYSTEM BARYCENTER': ICRS}
+
+    # Generate all SPICE frame classes
+    for class_num in _CLASS_TYPES.keys():
+        frames = spiceypy.kplfrm(class_num)
+        for frame_id in frames:
+            _install_frame_by_id(frame_id)
+
+
+@add_common_docstring(**_variables_for_parse_time_docstring())
+def get_body(body, time, *, spice_frame='J2000', observer=None):
+    """
+    Get the location of a body via SPICE.
+
+    Parameters
+    ----------
+    body : `int`, `str`
+        The NAIF body ID, or a string that is resolvable to a body ID
+    time : {parse_time_types}
+        Time to use in a parse_time-compatible format.
+    spice_frame : `str`
+        The SPICE frame name to use for the returned coordinate.  Defaults to
+        ``'J2000'``, which is equivalent to Astropy's `~astropy.coordinates.ICRS`.
+    observer : `~astropy.coordinates.SkyCoord`
+        If `None`, the returned coordinate is the instantaneous or “true” location.
+        If not `None`, the returned coordinate is the astrometric location (i.e.,
+        accounts for light travel time to the specified observer).
+    """
+    body_name = spiceypy.bodc2n(body) if isinstance(body, int) else body
+    obstime = parse_time(time)
+    et = _convert_to_et(obstime)
+
+    frame_center = spiceypy.frinfo(spiceypy.namfrm(spice_frame))[0]
+
+    if observer is None:
+        pos = spiceypy.spkpos(body_name,
+                              et,
+                              spice_frame,
+                              'NONE',
+                              spiceypy.bodc2n(frame_center))[0] << u.km
+    else:
+        obspos = observer.icrs.cartesian.xyz.to_value('km')
+        pos, lt = spiceypy.spkcpo(body_name,
+                                  et,
+                                  spice_frame,
+                                  'OBSERVER',
+                                  'CN',
+                                  obspos,
+                                  'SSB',
+                                  'J2000')
+        log.info(f"Apparent body location accounts for {lt:.2f} seconds of light travel time")
+
+        pos = pos[:3] << u.km
+        if spice_frame != 'J2000':
+            shift = spiceypy.spkpos(spiceypy.bodc2n(frame_center),
+                                    et,
+                                    'J2000',
+                                    'NONE',
+                                    'SSB')[0]
+            obspos -= shift
+
+            matrix = spiceypy.pxform('J2000', spice_frame, _convert_to_et(obstime))
+            obspos = matrix @ obspos
+        pos += obspos << u.km
+
+    frame_name = 'icrs' if spice_frame == 'J2000' else _astropy_frame_name(spice_frame)
+
+    return SkyCoord(CartesianRepresentation(pos.T), frame=frame_name, obstime=obstime)
+
+
+@add_common_docstring(**_variables_for_parse_time_docstring())
+def get_fov(instrument, time, *, resolution=100):
```

Comment:
Could [astropy's regions](https://astropy-regions.readthedocs.io/en/stable/index.html) package be useful here?

Reply:
Er, perhaps, marginally, at least for circular FOVs, but wouldn't technically do rectangular FOVs correctly.  Besides, `regions` hasn't had a release for over a year; is it something we'd want to depend on?

Reply:
Yeah I am not sure what the status of it is, I can ask around. Maybe we should look at what we would want it to do to support this?

Reply:
If you can ask around, we want to use it for the HEK client in the future.